### DISCUSSION TITLE
Add setting to not show a library on home page

### DISCUSSION THREAD

**Discussion by smorkin** [gotson/komga#1560] [Ideas]
### Describe your suggested feature

I'd like to be able to not include comics from a given library on the home page since it is a library which I use for temporary stuff and which changes very often → comics from this library are all that's shown in the 'Recently added' sections, drowning out all the more relevant stuff.

So something like a switch in the settings for a library, like "Include in home page" with default value to TRUE since the feature is definitely not something that all users would like for their new libraries.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the latest version.
- [X] I will fill out all of the requested information in this form.

**Comment by jonaballen** [comment]
I started writing up a request but found this, so adding here instead:

**Ability to include or exclude libraries from being included in the Home page content**
Some libraries might be unnecessary or unsuitable to show on Home, content could be NSFW or for some other reasons preferred to be excluded. 
As a user it would be nice to be able to exclude specific libraries from being shown on Home page since it is the starting point and universally accessible from Logo. 

Suggestion:
Add a setting in Edit Library > Options
"Exclude on Home screen" Info tooltip: "Books and Series will NOT be visible in the Home screen sections" that is disabled by default

**Comment by Stoffel324** [comment]
I would also like this feature.

My use case is that I have Magazines and Newspaper Libraries that I would like to exclude from my home page. These are more of an archive than an active library that I would read.

**Comment by kayhos** [comment]
I would also like this feature, previous persons who wanted this feature also have documented it in tickets https://github.com/gotson/komga/issues/229 and https://github.com/gotson/komga/issues/600. I created a request at #1634 , which was a dupe of this one.

Here are my use cases:
- A library in a language you do not understand. I'm hosting a library for a friend that's in chinese, I don't speak it and I'm not interested in the contents of his library.
- NSFW libraries. I have some series with explicit covers that I would rather my kids don't see when accessing books with them.
- Libraries that have content that is not pertinent or interesting to you. I have a library hosting history books for another friend, I don't care about his books and don't want to see them on my dashboard even though he loves his books.
- Reduce the content shown on the home page on very big Komga servers. As per my previous use cases, I have a bunch of libraries and would prefer to keep the dashboard focused on weekly content like mangas and comics instead of Work and other references.

**Comment by gotson** [comment]
I have added support to pin/unpin libraries, like Plex does.

The dashboard will only show content from pinned libraries. The global libraries view will show content (series/collections/readlists) from pinned libraries only, and if no library is pinned, from all libraries.

Unpinned libraries will be listed under a _More_ sublist:
<img width="257" alt="image" src="https://github.com/user-attachments/assets/d3dc5c2b-6106-438e-a3da-a321bd6f7cae" />

Pin/Unpin is accessible from the 3-dots menu:
<img width="271" alt="image" src="https://github.com/user-attachments/assets/1a3f127f-f5cd-48f0-8b98-6e3ccd2c301e" />

**Reply by gotson** [reply]
🎉 This issue has been resolved in `1.20.0` ([Release Notes](https://github.com/gotson/komga/releases/tag/1.20.0))

**Comment by smorkin** [comment]
tested & works fine in 1.20.0; thanks!
